### PR TITLE
OSVR HMD plugin

### PR DESCRIPTION
**Pull request by ctrlaltdavid**

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Build console output for commit: c39a7eb5a693dca943d04268a3389fbb9c8e792a available at: https://gist.github.com/6c59a162b7d55a9a1f733bed136c2885<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8320-c39a7eb.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8320-c39a7eb.dmg

File: cmake/externals/boost/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+include(SelectLibraryConfigurations)
+
+set(EXTERNAL_NAME boost)
+
+string(TOUPPER ${EXTERNAL_NAME} EXTERNAL_NAME_UPPER)
+
+if (WIN32)
+  ExternalProject_Add(
+    ${EXTERNAL_NAME}
+    URL http://ctrlaltstudio.com/downloads/hifi/boost_win_1_57_0.zip
```
Comment (jherico) [inline cmake/externals/boost/CMakeLists.txt]: Does OSVR seriously _require_ Boost libraries?  That's insane.

File: interface/CMakeLists.txt [hunk 2]
```diff
@@ -126,6 +126,8 @@ if (WIN32)
   add_dependency_external_projects(sdl2)
   add_dependency_external_projects(OpenVR)
   add_dependency_external_projects(neuron)
+  add_dependency_external_projects(osvr)
+  add_dependency_external_projects(boost)
```
Comment (jherico) [inline interface/CMakeLists.txt]: we should add a new kind of 'dependency' here that says "I depend on the dlls for this project, but not on the libs or includes".  Not a blocker for this PR though.

File: cmake/externals/boost/CMakeLists.txt (hunk 1 above)
Comment (ctrlaltdavid) [inline cmake/externals/boost/CMakeLists.txt]: Yes, it requires Boost: plug-in doesn't compile without.

**Comment by hifi-gustavo** [comment]
Build console output for commit: c39a7eb5a693dca943d04268a3389fbb9c8e792a available at: https://gist.github.com/6c59a162b7d55a9a1f733bed136c2885<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8320-c39a7eb.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8320-c39a7eb.dmg

**Comment by ctrlaltdavid** [comment]
Waiting on OSVR HMDs for final coding then testing.

**Comment by ctrlaltdavid** [comment]
Test this please

**Comment by hifi-gustavo** [comment]
Build console output for commit: dacb75509e493b00bae6852f5f07415be62db551 available at: https://gist.github.com/6c59a162b7d55a9a1f733bed136c2885<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8320-dacb755.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8320-dacb755.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: dacb75509e493b00bae6852f5f07415be62db551 available at: https://gist.github.com/4bf29f6ca26d1b96bfb8612e63311880<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8320-dacb755.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8320-dacb755.dmg

**Comment by ctrlaltdavid** [comment]
Test this please

**Comment by hifi-gustavo** [comment]
Build console output for commit: dacb75509e493b00bae6852f5f07415be62db551 available at: https://gist.github.com/6c59a162b7d55a9a1f733bed136c2885<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8320-dacb755.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8320-dacb755.dmg

**Comment by ctrlaltdavid** [comment]
Test this please

**Comment by hifi-gustavo** [comment]
Build console output for commit: dacb75509e493b00bae6852f5f07415be62db551 available at: https://gist.github.com/6c59a162b7d55a9a1f733bed136c2885<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8320-dacb755.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8320-dacb755.dmg

**Comment by ctrlaltdavid** [comment]
Test this please

**Comment by hifi-gustavo** [comment]
Build console output for commit: 944bdd764edeaaf16b67960937db46116d5c3df1 available at: https://gist.github.com/6c59a162b7d55a9a1f733bed136c2885<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8320-944bdd7.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8320-944bdd7.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 944bdd764edeaaf16b67960937db46116d5c3df1 available at: https://gist.github.com/6c59a162b7d55a9a1f733bed136c2885<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8320-944bdd7.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8320-944bdd7.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: ae47af8a0bb6aba6a70b253cb2a0785761c55ed4 available at: https://gist.github.com/6c59a162b7d55a9a1f733bed136c2885<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8320-ae47af8.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8320-ae47af8.dmg

**Comment by ctrlaltdavid** [comment]
Test this please

**Comment by hifi-gustavo** [comment]
Build console output for commit: ae47af8a0bb6aba6a70b253cb2a0785761c55ed4 available at: https://gist.github.com/6c59a162b7d55a9a1f733bed136c2885<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8320-ae47af8.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8320-ae47af8.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 24861d622a9ae51a3e0a082d68d6d66f0508a3f3 available at: https://gist.github.com/6c59a162b7d55a9a1f733bed136c2885<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8320-24861d6.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8320-24861d6.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 24861d622a9ae51a3e0a082d68d6d66f0508a3f3 available at: https://gist.github.com/6c59a162b7d55a9a1f733bed136c2885<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8320-24861d6.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8320-24861d6.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 198a8f2db74134c72b19e300e1a3c6f514832a08 available at: https://gist.github.com/6c59a162b7d55a9a1f733bed136c2885<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8320-198a8f2.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8320-198a8f2.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 727383091a8f54875bd06184f7c6887088ad0df9 available at: https://gist.github.com/6c59a162b7d55a9a1f733bed136c2885<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8320-7273830.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8320-7273830.dmg

**Comment by ctrlaltdavid** [comment]
Test this please

**Comment by hifi-gustavo** [comment]
Build console output for commit: 727383091a8f54875bd06184f7c6887088ad0df9 available at: https://gist.github.com/6c59a162b7d55a9a1f733bed136c2885<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8320-7273830.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8320-7273830.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 89c1d445323a15ec8895ba1f19d44b68615ba4dd available at: https://gist.github.com/6c59a162b7d55a9a1f733bed136c2885<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8320-89c1d44.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8320-89c1d44.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 89c1d445323a15ec8895ba1f19d44b68615ba4dd available at: https://gist.github.com/6c59a162b7d55a9a1f733bed136c2885<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8320-89c1d44.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8320-89c1d44.dmg

**Comment by ctrlaltdavid** [comment]
Test this please

**Comment by hifi-gustavo** [comment]
Build console output for commit: 89c1d445323a15ec8895ba1f19d44b68615ba4dd available at: https://gist.github.com/6c59a162b7d55a9a1f733bed136c2885<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8320-89c1d44.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8320-89c1d44.dmg

**Pull request closed without merging**